pkg/kube/controllers/boshdeployment/deployment_reconciler.go

```diff
@@ -217,3 +239,203 @@ func (r *ReconcileBOSHDeployment) createQuarksJob(ctx context.Context, instance
 
 	return err
 }
+
+// listLinkInfos returns a LinkInfos containing link providers if needed
+// and updates `quarks_links` properties
+func (r *ReconcileBOSHDeployment) listLinkInfos(instance *bdv1.BOSHDeployment, manifest *bdm.Manifest) (converter.LinkInfos, error) {
```

Comment:
I'll say more about this in the next retro, but I think we're all putting too much code in the 'reconciler.go' files.
The funcs which don't need a context, could often live with the structs they modify. 
The funcs that use a context could be separate interfaces, like the job_creator.go in quarks-job.

Reply:
A good point, thx.